app/controllers/categories_controller.rb

```diff
@@ -0,0 +1,48 @@
+class CategoriesController < ApplicationController
+  before_action :set_category, only: %i[show edit update destroy]
+
+  def index
+    @categories = Category.roots
+  end
+
+  def show; end
+
+  def new
+    @category = Category.new
+  end
+
+  def edit; end
+
+  def create
+    @category = Category.new(category_params)
+
+    if @category.save
+      redirect_to @category
+    else
+      render :new
+    end
+  end
+
+  def update
+    if @category.update(category_params)
+      redirect_to @category
+    else
+      render :edit
+    end
+  end
+
+  def destroy
+    @category.destroy
+    redirect_to categories_url
+  end
+
+  private
+
```

Comment:
Think about the creating helper for this code. As it's the same as here:
https://github.com/cybergizer/nicher/pull/26/files#diff-dbc6fdbfa3706e8655ca693ae7639d6bR54